homeassistant/components/spaceapi.py

```diff
@@ -0,0 +1,171 @@
+
+URL_API_SPACEAPI = '/api/spaceapi'
+
+LOCATION_SCHEMA = vol.Schema({
+    vol.Optional(CONF_ADDRESS): cv.string,
+}, required=True)
+
+CONTACT_SCHEMA = vol.Schema({
+    vol.Optional(CONF_EMAIL): cv.string,
+    vol.Optional(CONF_IRC): cv.string,
+    vol.Optional(CONF_MAILING_LIST): cv.string,
+    vol.Optional(CONF_PHONE): cv.string,
+    vol.Optional(CONF_TWITTER): cv.string,
+}, required=False)
+
+STATE_SCHEMA = vol.Schema({
+    vol.Required(CONF_ENTITY_ID): cv.entity_id,
+    vol.Optional(CONF_ICON_CLOSED): cv.string,
+    vol.Optional(CONF_ICON_OPEN): cv.string,
+}, required=False)
+
+SENSOR_SCHEMA = vol.Schema(
+    {vol.In(SENSOR_TYPES): [cv.entity_id]}
+)
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_CONTACT): CONTACT_SCHEMA,
+        vol.Required(CONF_ISSUE_REPORT_CHANNELS):
+            vol.All(cv.ensure_list, [vol.In(ISSUE_REPORT_CHANNELS)]),
+        vol.Required(CONF_LOCATION): LOCATION_SCHEMA,
+        vol.Required(CONF_LOGO): cv.string,
+        vol.Required(CONF_SPACE): cv.string,
+        vol.Required(CONF_STATE): STATE_SCHEMA,
+        vol.Required(CONF_URL): cv.string,
+        vol.Optional(CONF_SENSORS): SENSOR_SCHEMA,
+    }),
+}, extra=vol.ALLOW_EXTRA)
+
+
+def setup(hass, config):
+    """Register the SpaceAPI with the HTTP interface."""
+    hass.data[DATA_SPACEAPI] = config[DOMAIN]
+    hass.http.register_view(APISpaceApiView)
+
+    return True
+
+
+class APISpaceApiView(HomeAssistantView):
+    """View to provide details according to the SpaceAPI."""
+
+    url = URL_API_SPACEAPI
+    name = 'api:spaceapi'
+
+    @ha.callback
+    def get(self, request):
+        """Get SpaceAPI data."""
+        spaceapi = dict(request.app['hass'].data[DATA_SPACEAPI])
+        is_sensors = spaceapi.get('sensors')
+
+        location = {
+            ATTR_ADDRESS: spaceapi[ATTR_LOCATION][CONF_ADDRESS],
+            ATTR_LATITUDE: request.app['hass'].config.latitude,
+            ATTR_LONGITUDE: request.app['hass'].config.longitude,
+        }
+
+        state_entity = spaceapi['state'][ATTR_ENTITY_ID]
+        state = request.app['hass'].states.get(state_entity)
+        last_changed = request.app['hass'].states.get(
+            state_entity).last_updated
+
+        state = {
+            ATTR_ICON: {
+                ATTR_OPEN: spaceapi['state'][CONF_ICON_OPEN],
+                ATTR_CLOSE: spaceapi['state'][CONF_ICON_CLOSED],
+            },
+            ATTR_OPEN: bool(state),
+            ATTR_LASTCHANGE: dt_util.as_timestamp(last_changed),
+        }
+
+        if is_sensors is not None:
+            sensors = {}
+            for sensor_type in SENSOR_TYPES:
+                sensors[sensor_type] = []
+                for sensor in spaceapi['sensors'][sensor_type]:
+                    sensor_state = request.app['hass'].states.get(sensor)
+                    name = sensor_state.attributes[ATTR_FRIENDLY_NAME]
+                    unit = sensor_state.attributes[ATTR_UNIT_OF_MEASUREMENT]
```

Comment:
Not all entities have units of measurements set in their attributes.

To be fair, probably all temperature and humidity sensor integrations do set a unit of measurement. But if a user uses a template sensor they might not explicitly set a unit of measurement.

Probably not an issue, but pointing it out if it's something that should be fixed.

Reply:
It will become an issue if we start to implement more sensors. For the first implementation we should simply assume that there is an unit of measurement as it's limited to temperature and humidity.